zerver/lib/message.py

```diff
@@ -319,8 +320,10 @@ def render_markdown(message, content, realm_id=None, realm_alert_words=None, mes
         message.alert_words = set()
         message.links_for_preview = set()
 
-        if realm_id is None:
-            realm_id = message.sender.realm_id
+        if realm is None:
+            realm = message.get_realm()
+        realm_id = realm.id
```

Comment:
This change might introduce a performance regression; the old code didn't require the Message object to have been fetched with `select_related("sender")` to avoid an extra database query, and the new code does.

In practice, I think we already do the appropriate select_related everywhere, but just wanted to point this out.